source/Halibut.Tests/AsyncServicesFixture.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut.Tests.Support;
+using Halibut.Tests.Support.TestAttributes;
+using Halibut.Tests.Support.TestCases;
+using Halibut.Tests.TestServices.Async;
+using Halibut.Tests.Util;
+using Halibut.TestUtils.Contracts;
+using Halibut.Util;
+using NUnit.Framework;
+
+namespace Halibut.Tests
+{
+    public class AsyncServicesFixture : BaseTest
+    {
+        [Test]
+        [LatestClientAndLatestServiceTestCases(testAsyncServicesAsWell: true, testNetworkConditions: false)]
+        public async Task AsyncServiceWithParams_CanBeRegisteredAndResolved(ClientAndServiceTestCase clientAndServiceTestCase)
+        {
+            var clientAndServiceBuilder = clientAndServiceTestCase.CreateTestCaseBuilder()
+                .AsLatestClientAndLatestServiceBuilder();
+
+            if (clientAndServiceTestCase.ServiceAsyncHalibutFeature.IsEnabled())
+            {
+                clientAndServiceBuilder = clientAndServiceBuilder
+                    .WithAsyncService<IEchoService, IAsyncEchoService>(() => new AsyncEchoService());
+            }
+            else
+            {
+                clientAndServiceBuilder = clientAndServiceBuilder
```

Comment:
Is it valuable to have this if? would it make sense to only test async and so have only the `.WithAsyncService<IEchoService, IAsyncEchoService>` line?

Reply:
The main value in having the sync version is that it verifies that the behaviour still works like it did before in that scenario, which I think is worthwhile here as the feature is still behind a feature toggle.

Reply:
is the existing behaviour tested over in the UsageFixture?

Reply:
Yeah that's true.

Is there a way we can test _only_ the async service code path, or should I update the test attribute i.e. `testAsyncServicesAsWell`

Reply:
yes, `, testSyncService: false`